agent/lib_composer.c

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "php_agent.h"
+#include "fw_hooks.h"
+#include "fw_support.h"
+#include "nr_txn.h"
+#include "util_logging.h"
+#include "util_memory.h"
+#include "util_syscalls.h"
+
+static bool nr_execute_handle_autoload_composer_is_initialized() {
+  zend_class_entry* zce = NULL;
+
+  if (NULL == (zce = nr_php_find_class("composer\\installedversions"))) {
+    nrl_verbosedebug(NRL_INSTRUMENT,
+                     "Composer\\InstalledVersions class not found");
+    return false;
+  };
+
+  // the class is found - there's hope!
+  if (NULL == nr_php_find_class_method(zce, "getinstalledpackages")
+      || NULL == nr_php_find_class_method(zce, "getversion")) {
+    nrl_verbosedebug(
+        NRL_INSTRUMENT,
+        "Composer\\InstalledVersions class found, but methods not found");
+    return false;
+  }
+
+  return true;
+}
+
+static int nr_execute_handle_autoload_composer_init(const char* vendor_path) {
+  char* code = NULL;
+  zval retval;
+  int result = -1;
+
+  if (nr_execute_handle_autoload_composer_is_initialized()) {
+    nrl_verbosedebug(NRL_INSTRUMENT, "%s: already initialized", __func__);
+    return NR_SUCCESS;
+  }
+
+#if 0
```

Comment:
What is this code block supposed to do if it's always false?

Reply:
The code in `#if 0` part is from the initial implementation of this feature. However I determined that all it does is what is in the `#else` part. I left this in for now so that I can easily switch the implementations to demonstrate that there's no difference in behavior regardless of the method used. The final version will only have the `#else` part.